code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -110,14 +108,20 @@
 					if(istype(L, /mob/living/simple_animal/shade) || istype(L, /mob/living/simple_animal/hostile/construct))
 						var/mob/living/simple_animal/M = L
 						if(M.health < M.maxHealth)
-							M.adjustHealth(-2)
-		if(corruption.len)
-			var/turf/T = pick_n_take(corruption)
-			corruption -= T
+							M.adjustHealth(-1)
+			CHECK_TICK
+	if(last_corrupt <= world.time)
+		var/list/validturfs = list()
+		for(var/T in circleviewturfs(src, 5))
 			if(istype(T, /turf/closed) || istype(T, /turf/open/floor/engine/cult) || istype(T, /turf/open/space) || istype(T, /turf/open/floor/plating/lava))
```

Comment:
It only runs every 5 seconds, 10 if it found no turfs on last iteration, so it ~should~ be okay